hadoop-project/pom.xml

```
@@ -2737,6 +2737,46 @@
         </dependencies>
       </dependencyManagement>
     </profile>
+    <profile>
+      <id>jdk17</id>
+      <activation>
+        <jdk>[17,)</jdk>
+      </activation>
+      <properties>
+        <maven.compiler.release>17</maven.compiler.release>
```

I think we should allow targeting different Java versions:

```suggestion
        <maven.compiler.release>${javac.version}</maven.compiler.release>
```